airflow/utils/file.py

```diff
@@ -356,5 +356,6 @@ def get_unique_dag_module_name(file_path: str) -> str:
     if isinstance(file_path, str):
         path_hash = hashlib.sha1(file_path.encode("utf-8")).hexdigest()
         org_mod_name = Path(file_path).stem
+        org_mod_name = org_mod_name.replace("-", "_")
```

Comment:
```suggestion
        org_mod_name = Path(file_path).stem.replace("-", "_")
```

Reply:
resolve: Use regex as a more general approach to replace special characters